src/Compilers/Core/Portable/Diagnostic/ProgrammaticSuppressionInfo.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Diagnostics
+{
+    /// <summary>
+    /// Contains information about the source of a programmatic diagnostic suppression produced by an <see cref="DiagnosticSuppressor"/>.
+    /// </summary>
+    internal sealed class ProgrammaticSuppressionInfo : IEquatable<ProgrammaticSuppressionInfo>
+    {
+        public ImmutableHashSet<(string Id, LocalizableString Justification)> Suppressions { get; }
+
+        internal ProgrammaticSuppressionInfo(ImmutableHashSet<(string Id, LocalizableString Justification)> suppressions)
+        {
+            Suppressions = suppressions;
+        }
+
+        public bool Equals(ProgrammaticSuppressionInfo other)
+        {
+            return other != null &&
+                this.Suppressions.SetEquals(other.Suppressions);
+        }
+
+        public override bool Equals(object obj)
+        {
+            return Equals(obj as ProgrammaticSuppressionInfo);
+        }
+
+        public override int GetHashCode()
+        {
+            var hash = Suppressions.Count.GetHashCode();
+            foreach (var suppression in Suppressions.Order())
```

Comment:
>.Order() [](start = 52, length = 8)

What is this method supposed to do? What would be the key for the ordering operation, where is it specified?

Reply:
This is just the default ordering to ensure deterministic hash code computation.

Reply:
> This is just the default ordering to ensure deterministic hash code computation.

What is the default ordering. Why is it good? Where is it defined?

---
In reply to: [292667893](https://github.com/dotnet/roslyn/pull/36067#discussion_r292667893) [](ancestors = 292667893)

Reply:
Default ordering of tuples in the set? I presume this would order the tuples by comparing each tuple element: https://docs.microsoft.com/en-us/dotnet/api/system.tuple-2.system-icomparable-compareto

Reply:
The extension method invoked is here: http://source.roslyn.io/#Microsoft.CodeAnalysis/InternalUtilities/EnumerableExtensions.cs,298